pkgs/development/python-modules/pycuda/default.nix

```diff
@@ -2,7 +2,6 @@
 , fetchPypi
 , fetchFromGitHub
 , Mako
-, boost
```

Comment:
Does our boost version no longer work with package?

Reply:
When I try to compile with boost, I get the following error:

```
g++ -pthread -shared -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-zlib-1.2.11/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-bzip2-1.0.6.0.1/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-expat-2.2.8/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xz-5.2.5/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libffi-3.3/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gdbm-1.18.1/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-sqlite-3.31.1/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-readline-6.3p08/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ncurses-6.2/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-openssl-1.1.1g/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-zlib-1.2.11/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-bzip2-1.0.6.0.1/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-expat-2.2.8/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xz-5.2.5/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libffi-3.3/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gdbm-1.18.1/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-sqlite-3.31.1/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-readline-6.3p08/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ncurses-6.2/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-openssl-1.1.1g/lib build/temp.linux-x86_64-3.7/src/cpp/cuda.o build/temp.linux-x86_64-3.7/src/cpp/bitlog.o build/temp.linux-x86_64-3.7/src/wrapper/wrap_cudadrv.o build/temp.linux-x86_64-3.7/src/wrapper/mempool.o build/temp.linux-x86_64-3.7/src/wrapper/wrap_curand.o -L/nix/store/693d3mxvg13xswg132mrjs6cg448mh7b-boost-1.69.0/lib -L/nix/store/187m9ag5m39f0hnhh0la6dbqjmygdp0p-cudatoolkit-10.2.89/lib -L/nix/store/187m9ag5m39f0hnhh0la6dbqjmygdp0p-cudatoolkit-10.2.89/lib64 -L/nix/store/187m9ag5m39f0hnhh0la6dbqjmygdp0p-cudatoolkit-10.2.89/lib/stubs -L/nix/store/187m9ag5m39f0hnhh0la6dbqjmygdp0p-cudatoolkit-10.2.89/lib64/stubs -L/nix/store/187m9ag5m39f0hnhh0la6dbqjmygdp0p-cudatoolkit-10.2.89/lib -L/nix/store/187m9ag5m39f0hnhh0la6dbqjmygdp0p-cudatoolkit-10.2.89/lib64 -L/nix/store/187m9ag5m39f0hnhh0la6dbqjmygdp0p-cudatoolkit-10.2.89/lib/stubs -L/nix/store/187m9ag5m39f0hnhh0la6dbqjmygdp0p-cudatoolkit-10.2.89/lib64/stubs -L/nix/store/2dcsn57cgaxs92ha5swihrab0g3l2h6g-python3-3.7.7/lib -lboost_python3 -lboost_thread -lcuda -lcurand -lpython3.7m -o build/lib.linux-x86_64-3.7/pycuda/_driver.cpython-37m-x86_64-linux-gnu.so
/nix/store/a57856fs4m8ir6vlv14h3gq3sv9aq2lb-binutils-2.31.1/bin/ld: cannot find -lboost_python3
collect2: error: ld returned 1 exit status
error: command 'g++' failed with exit status 1
builder for '/nix/store/m91kqwgwpdajb4mj0gpk2phhkp25s2l5-python3.7-pycuda-2019.1.2.drv' failed with exit code 1
error: build of '/nix/store/m91kqwgwpdajb4mj0gpk2phhkp25s2l5-python3.7-pycuda-2019.1.2.drv' failed
```

Reply:
Ah found the issue, boost_python37 works instead of boost_python, so I guess we have to pass the full python version to the linker now

https://github.com/inducer/pycuda/blob/e8f72360fc56955020680479a753dacc57db722b/aksetup_helper.py#L528

Reply:
We might also make a symlink in our boost package if that is a more general problem. A different hack would be to add the symlink to this package i.e. `$out/lib/libboost_python3.so` and add the `NIX_LDFLAGS="-L $out/lib";`

Reply:
Actually there code look right on the first glance:

```
>>> "boost_python-py%d%d" % sys.version_info[:2]
'boost_python-py37'
```

Reply:
Right currently we're always calling 'boost_python-py3' instead of 'boost_python-py37', I'm trying to fix now but my nix is... passable

I tried doing
``` nix
 --boost-python-libname=boost_python${stdenv.lib.optionalString isPy3k 3${stdenv.lib.versions.minor python.version}}
```
But I'm guessing you can't nest like that?

Reply:
I'm not sure if this fails on python2 now since I removed the n${stdenv.lib.optionalString isPy3k} check, but the latest commit now builds with boost.